### PR TITLE
Document VotesExtended assumptions

### PR DESCRIPTION
**Pull request by Amxx**
Fixes L-02

### CODE REVIEW

File: contracts/governance/utils/VotesExtended.sol [hunk 1]
```diff
@@ -7,6 +7,28 @@ import {SafeCast} from "../../utils/math/SafeCast.sol";
 
 /**
  * @dev Extension of {Votes} that adds exposes checkpoints for delegations and balances.
+ *
+ * WARNING: While this contract extends {Votes}, it makes additional assumption. Valid usage of {Votes} may not work
```
Comment (Amxx) [inline contracts/governance/utils/VotesExtended.sol:11]:
```suggestion
 * WARNING: While this contract extends {Votes}, it makes additional assumptions. Valid uses of {Votes} may not work
```

File: contracts/governance/utils/VotesExtended.sol [hunk 2]
```diff
@@ -7,6 +7,28 @@ import {SafeCast} from "../../utils/math/SafeCast.sol";
 
 /**
  * @dev Extension of {Votes} that adds exposes checkpoints for delegations and balances.
+ *
+ * WARNING: While this contract extends {Votes}, it makes additional assumptions. Valid uses of {Votes} may not be
+ * compatbile with {VotesExtended} without additional considerations. The implementation of {_transferVotingUnits}
```
Comment (Amxx) [inline contracts/governance/utils/VotesExtended.sol:12]:
```suggestion
 * compatible with {VotesExtended} without additional considerations. The implementation of {_transferVotingUnits}
```

File: contracts/governance/utils/VotesExtended.sol [hunk 3]
```diff
@@ -7,6 +7,28 @@ import {SafeCast} from "../../utils/math/SafeCast.sol";
...
+ * expects to run AFTER the voting weight movement is registered in a way that would reflect on {_getVotingUnits}.
+ *
+ * Said differently, {VotesExtended} MUST be integrated in a way such that calls {_transferVotingUnits} AFTER the 
+ * asset transfer is registered adn balances are updated:
```
Comment (Amxx) [inline contracts/governance/utils/VotesExtended.sol:16]:
```suggestion
 * asset transfer is registered and balances are updated:
```

File: contracts/governance/utils/VotesExtended.sol [hunk 4]
```diff
@@ -7,6 +7,28 @@ import {SafeCast} from "../../utils/math/SafeCast.sol";
 
 /**
  * @dev Extension of {Votes} that adds exposes checkpoints for delegations and balances.
+ *
+ * WARNING: While this contract extends {Votes}, it makes additional assumptions. Valid uses of {Votes} may not be
+ * compatible with {VotesExtended} without additional considerations. The implementation of {_transferVotingUnits}
+ * expects to run AFTER the voting weight movement is registered in a way that would reflect on {_getVotingUnits}.
```
Comment (ernestognw) [inline contracts/governance/utils/VotesExtended.sol:13]: I would simplify it to:

```suggestion
* WARNING: While this contract extends from {Votes}, the implementation of {_transferVotingUnits} must run 
* after any state change that affects the result of {_getVotingUnits}. Consider that the order of execution may
* cause inconsistent unit tracking.
```

Comment (Amxx) [reply contracts/governance/utils/VotesExtended.sol:13]: "extends from {Votes}" doesn't sound right to me. We should get a native english speaker to double check. @arr00

Comment (ernestognw) [reply contracts/governance/utils/VotesExtended.sol:13]: I'm fine with "extends from" too

Comment (arr00) [reply contracts/governance/utils/VotesExtended.sol:13]: WARNING: While this contract extends {Votes}, valid uses of {Votes} may not be compatible with {VotesExtended}, without additional considerations. This implementation of {_transferVotingUnits} must run AFTER the voting weight movement is registered, such that it is reflected on {_getVotingUnits}.

Agree with @Amxx that "extends {Votes}" is preferred. I modified it slightly to not repeat the idea of changes to the flow of `_transferVotingUnits`.

File: contracts/governance/utils/VotesExtended.sol [hunk 5]
```diff
@@ -7,6 +7,28 @@ import {SafeCast} from "../../utils/math/SafeCast.sol";
 
 /**
  * @dev Extension of {Votes} that adds checkpoints for delegations and balances.
+ *
+ * WARNING: While this contract extends {Votes}, valid uses of {Votes} may not be compatible with 
+ * {VotesExtended} without additional considerations. This implementation of {_transferVotingUnits} must
+ * run AFTER the voting weight movement is registered, such that it is reflected on {_getVotingUnits}.
+ *
+ * Said differently, {VotesExtended} MUST be integrated in a way such that calls {_transferVotingUnits} AFTER the
```
Comment (Amxx) [inline contracts/governance/utils/VotesExtended.sol:15]:
```suggestion
 * Said differently, {VotesExtended} MUST be integrated in a way that calls {_transferVotingUnits} AFTER the
```

**Review by ernestognw — APPROVED** [review]

**Pull request merged**